dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/HostAPIImpl.java

```diff
@@ -884,51 +883,23 @@ public Host DBSearch(String id, User user, boolean respectFrontendRoles) throws
         if (!UtilMethods.isSet(id))
             return null;
 
-        final String languageIdColumn = "language_id";
-        final String isDefaultColumn = "isDefault";
-
+        Host host = null;
 
-        List<Field> fields = hostType().fields();
+        final ContentletVersionInfo vinfo = HibernateUtil.load(ContentletVersionInfo.class,
+                "from "+ContentletVersionInfo.class.getName()+" where identifier=?", id);
 
-        StringBuilder sql = new StringBuilder();
-        sql.append("SELECT inode," + languageIdColumn);
-        for (Field field: fields) {
-            if (field.dataType() != DataTypes.SYSTEM) {
-                sql.append(", " + field.variable());
-            }
-        }
-        sql.append(" FROM host ");
-        sql.append("WHERE identifier='" + id + "'");
-        
-        SQLQueryFactory sqlQueryFactory = new SQLQueryFactory(sql.toString());
-        Query query = sqlQueryFactory.getQuery();
+        if(vinfo!=null && UtilMethods.isSet(vinfo.getIdentifier())) {
 
-        List<Map<String, Serializable>> list = APILocator.getContentletAPI().DBSearch(query, user, respectFrontendRoles);
-        if (1 < list.size())
-            Logger.error(this, "More of one working version of host match the same identifier " + id + "!!");
-        else if (list.size() == 0)
-            return null;
+            User systemUser = APILocator.systemUser();
 
-        Host host = new Host();
-        
-        for (String key: list.get(0).keySet()) {
-            Object value = list.get(0).get(key);
-            if ( key.equals(languageIdColumn) ) {
-                if ( value instanceof Number){ //Hibernate maps Oracle NUMBER to BigDecimal.
-                    host.setProperty(Contentlet.LANGUAGEID_KEY, ((Number) value).longValue());
-                } else {
-                    host.setProperty(Contentlet.LANGUAGEID_KEY, value);
-                }
-            } if (key.equals(isDefaultColumn)) { 
-                host.setProperty(Host.IS_DEFAULT_KEY, DbConnectionFactory.isDBTrue(value.toString()));
-            } else {
-                host.setProperty(key, value);
+            String hostInode=vinfo.getWorkingInode();
+            final Contentlet cont= APILocator.getContentletAPI().find(hostInode, systemUser, respectFrontendRoles);
+            final ContentType type =APILocator.getContentTypeAPI(systemUser, respectFrontendRoles).find(Host.HOST_VELOCITY_VAR_NAME);
+            if(cont.getStructureInode().equals(type.inode())) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this use of "getStructureInode"; it is deprecated. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)